src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -292,6 +292,7 @@ TEST_F(RetrieveInstancesWidgetTest, StartSuccessfulDefault) {
 
   widget_.Start();
   QCoreApplication::processEvents();
+  QCoreApplication::processEvents();
```

Comment:
Twice? Something should be wrong here.

Reply:
No, its actually not, but I added a comment why this is necessary.

Reply:
But shouldn't this be different? I still find it really strange. @beckerhe

Reply:
There 2 ways to solve these a "queued event queues another event" issues. The first is what I did here, call processEvents() as often as that happens. Which is not ideal, because if another action like this is added, then another processEvents() would need to be added and so on. The alternative is using a QTimer and stop the application after a certain time (5ms or so). This is also not ideal, because it adds this timing component, which can potentially cause flakiness. 

I decided here for the first option, being aware that its not a perfect solution, that might need to be changed in the future again